### PR TITLE
add codecov-cli and deps

### PR DESCRIPTION
**Pull request by giovanni-guidini**
These changes were made by following the exact instructions on Readme.md, that is, running `python3 -m add_pkg codecov-cli`.

This is used in the `codecov-ats.yml` workflow and currently requires the extra index to Pypi to be installed.
See: https://github.com/getsentry/sentry/blob/d069119cee5dfc887cf3f0744e4c8ebdf9be8a9c/.github/workflows/codecov_ats.yml#L76C100-L76C100

### CODE REVIEW

File: packages.ini
Comment (asottile-sentry) [reply packages.ini:479]: actually ijson is dead and archived too so we definitely should not be using it

for a few 100 MB I doubt it will make a performance difference to just use stdlib json

Comment (asottile-sentry) [reply packages.ini:479]: oh maybe not archived -- just the pypi url goes to the old maintainer

Comment (asottile-sentry) [reply packages.ini:479]: to be clear here -- ijson is _probably_ fine -- I don't have major concerns with it but it is unlikely to be solving the problem at hand.  `smart-open` isn't getting you anything though -- normal `open` is already streamed unless you call `.read()` on it

**Comment by giovanni-guidini** [comment]
Blocked by https://github.com/codecov/codecov-cli/pull/326 (to remove `smart-open`) and pending a new release being made

**Review by bukzor-sentryio — CHANGES_REQUESTED** [review]
just need to remove smart-open

**Pull request closed without merging**
